packages/web-media-box/src/dash-parser/parse.ts

```diff
@@ -1,19 +1,66 @@
-import { ParsedManifest } from "./types/parsedManifest";
+import { Attributes, ParsedManifest } from "./types/parsedManifest";
 import { testString } from "./examples/mpd";
+import { handleNode } from "./handleNode";
+import { parseAttributes } from "./parseAttributes";
 
 /**
  * Parses a MPD manifest file.
  *
- * @param playlist The URL of the mpd manifest to be parsed.
+ * @param manifest A string containing the XML of a DASH manifest.
  */
-export default function parse(playlist: string): ParsedManifest {
-  const parsedManifest: ParsedManifest = {
-    segments: [],
-    custom: {}
+export default function parse(manifest: string, url: string): ParsedManifest {
+  const initialParsedManifest: ParsedManifest = {
+    uri: url || '',
+    representations: [],
   };
 
-  // TODO: implement parsing.
-  var doc = new DOMParser().parseFromString(testString, 'text/xml');
+  const doc = new DOMParser().parseFromString(manifest, 'text/xml');
 
-  return parsedManifest;
+  return iterateXMLNodes(doc, initialParsedManifest);
```

Comment:
nit: a semantically right design for the iterate function would be just iteration, for node processing pass the callback, it should not care about manifest object and how it should be processed